### ISSUE TITLE
regression: most magit operations seem to cause lots of file reverts

### ISSUE THREAD

**Issue by vasaitis** [magit/magit#1856]
I run Emacs with global-auto-revert-mode, and I use magit. A few days ago I upgraded to the most recent version (from MELPA I suppose?), magit-20150513.457, and since then *a lot* of magit operations will cause Emacs to revert all buffers containing open files from the repository that is being acted on. Notable examples are magit-refresh, and also magit-toggle-section on a line representing a file's (un)staged changes (show/hide a file's hunks).

It's not the end of the world, but it is *very* annoying, and I don't understand why operations like that would cause the files to be constantly modified (and if they're not, why the operations would cause Emacs to revert them). If you could fix this, I'd be very grateful!

**Comment by tarsius** [comment]
> I don't understand why operations like that would cause the files to be constantly modified

They should not. And if they do then that would be a bug but I strongly doubt the mentioned commands change any files. Have you *verified* that these commands change the files, i.e. checked the modification times before and after invoking such a command?

> (and if they're not, why the operations would cause Emacs to revert them).

There might be a bug in *Emacs*. See this commit in the Emacs repository: 6444482c63b8f148c44904f11599de643e903dca. If I understand that correctly then, prior to this fix, the global mode did revert all file-visiting buffers every five seconds. To get the fix update to 24.5.

**Comment by vasaitis** [comment]
> Have you *verified* that these commands change the files, i.e. checked the modification times before and after invoking such a command?

Indeed the files don't actually change, I just checked.

> There might be a bug in *Emacs*. See this commit in the Emacs repository: 6444482c63b8f148c44904f11599de643e903dca. If I understand that correctly then, prior to this fix, the global mode did revert all file-visiting buffers every five seconds. To get the fix update to 24.5.

Perhaps, I can check this as soon as 24.5 hits Homebrew. But it's still the case that I only observe unnecessary reverts in combination with magit operations, and this only started happening when I last updated magit.

**Comment by dunn** [comment]
24.5 is in Homebrew!

**Comment by tarsius** [comment]
Turns out that's a regression in Magit. I have fixed it now.

But it has little to do with `global-auto-revert-mode`. Magit has its own revert mode and the regression happened when I stopped reusing code from `autorevert.el` to implement that.

By the way, using both modes at the same time should not cause any problems.

**Comment by vasaitis** [comment]
> 24.5 is in Homebrew!

Silly me, I did in fact have 24.5 in my system, but I was launching Emacs via the `/Applications` link created via `brew linkapps`, and that hadn't been updated for whatever reason. (Homebrew seems to have been fixed now to point these to `/usr/local/opt`, so that they don't need to be updated.) And now that I fixed that I discovered I also needed to fix the fact that the Dock still pointed to the previous version regardless. D'oh!

Anyway, I can confirm that I no longer get all these redundant reverts, thanks!
